packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.styles.ts

```diff
@@ -0,0 +1,294 @@
+import { IDropdownStyles, IDropdownStyleProps } from './Dropdown.types';
+import { IStyleFunction } from '../../Utilities';
+import {
+  FontSizes,
+  FontWeights,
+  HighContrastSelector,
+  IRawStyle,
+  IStyle,
+  getGlobalClassNames,
+  normalize
+} from '../../Styling';
+
+const GlobalClassNames = {
+  root: 'ms-Dropdown-container',
+  label: 'ms-Dropdown-label',
+  dropdown: 'ms-Dropdown',
+  title: 'ms-Dropdown-title',
+  caretDownWrapper: 'ms-Dropdown-caretDownWrapper',
+  caretDown: 'ms-Dropdown-caretDown',
+  callout: 'ms-Dropdown-callout',
+  panel: 'ms-Dropdown-panel',
+  dropdownItems: 'ms-Dropdown-items',
+  dropdownDivider: 'ms-Dropdown-divider',
+  dropdownOptionText: 'ms-Dropdown-optionText',
+  dropdownItemHeader: 'ms-Dropdown-header',
+  titleIsPlaceHolder: 'ms-Dropdown-titleIsPlaceHolder',
+  titleHasError: 'ms-Dropdown-title--hasError'
+};
+
+const DROPDOWN_HEIGHT = 32;
+const DROPDOWN_ITEMHEIGHT = 32;
+
+const highContrastAdjustMixin = {
+  // highContrastAdjust mixin
+  '@media screen and (-ms-high-contrast: active), screen and (-ms-high-contrast: black-on-white)': {
+    MsHighContrastAdjust: 'none'
+  }
+};
+
+const highContrastItemAndTitleStateMixin: IRawStyle = {
+  selectors: {
+    [HighContrastSelector]: {
+      backgroundColor: 'Highlight',
+      borderColor: 'Highlight',
+      color: 'Highlight'
+    },
+    ...highContrastAdjustMixin
+  }
+};
+
+const highContrastBorderState: IRawStyle = {
+  selectors: {
+    [HighContrastSelector]: {
+      borderColor: 'Highlight'
+    }
+  }
+};
+
+export const getStyles: IStyleFunction<IDropdownStyleProps, IDropdownStyles> = props => {
+  const {
+    theme,
+    hasError,
+    className,
+    isOpen,
+    disabled,
+    required,
+    isRenderingPlaceholder,
+    panelClassName,
+    calloutClassName
+  } = props;
+
+  if (!theme) {
```

Comment:
Not sure how I feel about having a throw in here since theme is technically called out as an optional. I think you're trying to avoid doing `!`'s in the base.tsx, but actually if you think about it, since we're guaranteed the existence of theme in the component, it is perfectly legit there (that's why ! exists, it is to tell compiler that you know what you're doing at runtime, etc).

I'm kinda in favor of being consistent with the other components that have taken that approach rather than having to throw...

Reply:
I would really like to minimize the use of `!` to the fullest extent possible such that we justify every use of `!` (like at https://github.com/OfficeDev/office-ui-fabric-react/blob/c9f6cb390d3e572a3082f397636616809cd291b4/packages/merge-styles/src/mergeStyleSets.ts#L108) 

The problem with `!` is that it is _very_ easily abusable and makes it really hard to reason about code correctness when you start seeing blocks of code with multiple usages of `!`.

I think it is MUCH better to follow the defensive programming paradigm of expressing the expectation that theme is always defined by what is essentially a check/assert. This would be the type of code you would write to express the expectation in languages without explicit null/undefined types anyway such as Java/C#.